GH-2939: Update PULL_REQUEST_TEMPLATE (#2940)
apache/parquet-java: .github/PULL_REQUEST_TEMPLATE.md

<!--
Thanks for opening a pull request!

> If this is your first pull request you can find detailed information on how 

Reviewer: If you're new to Parquet-Java, information on how to contribute can be found here: https://parquet.apache.org/docs/contribution-guidelines/contributing

to contribute here: [Contributing Overview](https://parquet.apache.org/docs/contribution-guidelines/contributing)

> If this is not a minor PR, could you open an issue for this pull request on GitHub? https://github.com/apache/parquet-java/issues/new/choose

Reviewer: ```suggestion
when contributing large features, open an issue for this pull request on GitHub:  https://github.com/apache/parquet-java/issues/new/choose
```

> Opening GitHub issues ahead of time contributes to the [Openness](http://theapacheway.com/open/#:~:text=Openness%20allows%20new%20users%20the,must%20happen%20in%20the%20open.) of the Apache Parquet-Java project.

Reviewer: ```suggestion
Opening GitHub issues ahead of time ensure openness of the Apache Parquet-Java project, read more about this here: http://theapacheway.com/open/#:~:text=Openness%20allows%20new%20users%20the,must%20happen%20in%20the%20open.
```

> Then could you also rename the pull request title in the following format?

Reviewer: ```suggestion
Please format pull request title in the following format:
```

-->

### What changes are included in this PR?

<!--
There is no need to duplicate the description in the issue here but it is sometimes worth providing a summary of the individual changes in this PR.
-->

### Are these changes tested?

<!--
We typically require tests for all PRs in order to:
1. Prevent the code from being accidentally broken by subsequent changes
2. Serve as another way to document the expected behavior of the code

If tests are not included in your PR, please explain why (for example, are they covered by existing tests)?
-->

### Are there any user-facing changes?

<!--
If there are user-facing changes then we may require documentation to be updated before approving the PR.
-->

<!--
Please uncomment the line below (and provide explanation) if you are adding a dependency. Check if the license complies with
the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
-->
<!-- **This PR adds new dependency.** -->

<!--
If there are any breaking changes to public APIs, please uncomment the line below and explain which changes are breaking.
-->
<!-- **This PR includes breaking changes to public APIs.** -->

> <!--

Reviewer: I might want to leave the critical fix out here. I dislike long PR templates, and I find that people don't really read them :'( Less is more I would say, or we should link to existing content (for example, link to the ASF Security page: https://www.apache.org/security/).
